paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationInterceptor.kt

```diff
@@ -143,18 +144,18 @@ internal class DefaultIntentConfirmationInterceptor @Inject constructor(
     override suspend fun intercept(
         initializationMode: PaymentSheet.InitializationMode,
         paymentMethod: PaymentMethod,
+        paymentMethodOptionsParams: PaymentMethodOptionsParams?,
         shippingValues: ConfirmPaymentIntentParams.Shipping?,
-        requiresSaveOnConfirmation: Boolean,
-        recollectedCvc: String?
     ): NextStep {
         return when (initializationMode) {
             is PaymentSheet.InitializationMode.DeferredIntent -> {
+                val offSession = ConfirmPaymentIntentParams.SetupFutureUsage.OffSession
                 handleDeferredIntent(
                     intentConfiguration = initializationMode.intentConfiguration,
                     paymentMethod = paymentMethod,
+                    paymentMethodOptionsParams = paymentMethodOptionsParams,
                     shippingValues = shippingValues,
-                    shouldSavePaymentMethod = requiresSaveOnConfirmation,
-                    recollectedCvc = recollectedCvc
+                    shouldSavePaymentMethod = paymentMethodOptionsParams?.setupFutureUsage() == offSession,
```

Comment:
I'm not 100% convinced this is needed, but it seemed safer to leave than to remove.